rpcserver.go

```diff
@@ -3204,6 +3231,93 @@ func (r *rpcServer) DeleteFederationServer(ctx context.Context,
 	return &unirpc.DeleteFederationServerResponse{}, nil
 }
 
+// SetFederationSyncConfig sets the configuration of the universe federation
+// sync.
+func (r *rpcServer) SetFederationSyncConfig(ctx context.Context,
+	req *unirpc.SetFederationSyncConfigRequest) (
+	*unirpc.SetFederationSyncConfigResponse, error) {
+
+	// Unmarshal general sync config.
+	var syncGeneralConfig *universe.FedGeneralSyncConfig
+	if req.GeneralSyncConfig != nil {
+		proofTypes, err := universe.NewSyncProofTypesFromString(
+			req.GeneralSyncConfig.ProofTypes,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse proof "+
+				"types: %w", err)
+		}
+		syncGeneralConfig = &universe.FedGeneralSyncConfig{
+			ProofTypes: proofTypes,
+		}
+	}
+
+	// Unmarshal asset (asset/asset group) specific sync configs.
+	assetSyncConfigs := make(
+		[]*universe.FedUniSyncConfig, len(req.AssetSyncConfig))
+	for i := range req.AssetSyncConfig {
+		assetSyncConfig := req.AssetSyncConfig[i]
+		config, err := unmarshalAssetSyncConfig(assetSyncConfig)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse asset sync "+
+				"config: %w", err)
+		}
+
+		assetSyncConfigs[i] = config
+	}
+
+	// Update asset (asset/asset group) specific sync configs.
+	err := r.cfg.FederationDB.UpsertFederationSyncConfig(
+		ctx, syncGeneralConfig, assetSyncConfigs,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to set federation sync "+
+			"config: %w", err)
+	}
+
+	return &unirpc.SetFederationSyncConfigResponse{}, nil
+}
+
+// QueryFederationSyncConfig queries the universe federation sync configuration
+// settings.
+func (r *rpcServer) QueryFederationSyncConfig(ctx context.Context,
+	_ *unirpc.QueryFederationSyncConfigRequest,
+) (*unirpc.QueryFederationSyncConfigResponse, error) {
+
+	// Obtain the general and universe specific federation sync configs.
+	generalConfig, uniSyncConfigs, err :=
+		r.cfg.FederationDB.QueryFederationSyncConfigs(ctx)
```

Comment:
Can use a temp variable to allow a proper line fold?

Reply:
Like this?
```
	queryFedSyncConfigs := r.cfg.FederationDB.QueryFederationSyncConfigs
	generalConfig, uniSyncConfigs, err := queryFedSyncConfigs(ctx)
```

Reply:
Yeah I think that work work?